feat: create project, add SpendingsInput component that is suitable f…

blohinv opened the pull request:
…or where spent and how much spent cases, make their validation

Commit:
feat: create project, add SpendingsInput component that is suitable for where spent and how much spent cases, make their validation

baklanovairina commented on src/components/AddButton/index.jsx:
```diff
@@ -0,0 +1,15 @@
+import "./style.scss";
+
+const AddButton = ({ action, name }) => {
```
там вроде целая форма, почему здесь только кнопка? 
какой тоне законченный функционал
и что action, что name? Действие и имя чего?

baklanovairina commented on src/components/AddButton/index.jsx:
```diff
@@ -0,0 +1,15 @@
+    <button
+      className="add-button"
+      type="button"
+      onClick={() => action()}
```
мы должны вызывать функцию, а не объявлять

baklanovairina commented on src/components/AddButton/index.jsx:
```diff
@@ -0,0 +1,15 @@
+
+const AddButton = ({ action, name }) => {
+  return (
+    <button
```
вообще создай UI компонент Button и передавай туда пропсы и чилдрен

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+const MainPage = () => {
+  const [spending, setSpending] = useState({
+    where: '',
+    howMuch: '',
```
тоже нейминг поправь, а то просто какие вопросительные слова и ничего не понятно

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+import "./style.scss";
+
+const MainPage = () => {
+  const [spending, setSpending] = useState({
```
здесь больше подойдет cost или expense

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+  const addSpending = () => {
+    if (validateInputs()) {
+      // localStorage.setItem(`spending-${spending.id}`, JSON.stringify(spending));
+      // setSpending({ ...spending, where: '', howMuch: '', id: uid() });
```
комментарии удалить

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+    }
+  }
+
+  const validateInputs = () => spending.where !== '' && spending.howMuch !== '';
```
не пишем функции в одну строку

кроме того, где нормальная валидация? Вывод ошибок? 
больше на черновик похоже

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+    <div className="container">
+      <h2 className="container__title">Учет моих расходов</h2>
+      <div className="container-add-spending">
+      <SpendingsInput
```
табы

baklanovairina replied:
и это должна быть отдельная форма-компонент
а кастомный инпут должен быть универсальным, а не чисто для расхода - Spendings

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+      <div className="container-add-spending">
+      <SpendingsInput
+        description={"Куда было потрачено"}
+        additionalName={"wide"}
```
зачем ты пишешь везде {} скобки передавая обычную строку??

baklanovairina replied:
и что это за пропс? для чего он?

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+        description={"Куда было потрачено"}
+        additionalName={"wide"}
+        inputHandler={inputHandler}
+        objKey="where"
```
это скорей name, а не objKey

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,54 @@
+      <SpendingsInput
+        description={"Куда было потрачено"}
+        additionalName={"wide"}
+        inputHandler={inputHandler}
```
handleChangeInput

baklanovairina commented on src/components/SpendingsInput/index.jsx:
```diff
@@ -0,0 +1,18 @@
+import "./style.scss";
+
+const SpendingsInput = ({ description, additionalName, inputHandler, objKey, value, textType }) => {
```
делай перенос строк

baklanovairina commented on src/components/SpendingsInput/index.jsx:
```diff
@@ -0,0 +1,18 @@
+    <div className="spendings-input">
+      <p className="spendings-input__description">{`${description}:`}</p>
+      <input
+        className={`spendings-input__value ${additionalName}`}
```
аааа additionalName - это класс
Так назови его классом и передавай как дополнительный класс

baklanovairina commented on src/components/SpendingsInput/index.jsx:
```diff
@@ -0,0 +1,18 @@
+const SpendingsInput = ({ description, additionalName, inputHandler, objKey, value, textType }) => {
+  return (
+    <div className="spendings-input">
+      <p className="spendings-input__description">{`${description}:`}</p>
```
description - это уже строка в пропсах, зачем ты ее снова преобразуешь?

baklanovairina commented on src/components/SpendingsInput/style.scss:
```diff
@@ -0,0 +1,35 @@
+.spendings-input {
+  display: flex;
+  flex-direction: column;
+  &__description {
```
отступы

baklanovairina commented on src/components/SpendingsInput/style.scss:
```diff
@@ -0,0 +1,35 @@
+    &::placeholder {
+      position: relative;
+      left: 15px;
+      font-family: "Roboto", sans-serif;
```
слишком часто повторяешься
задай его 1 раз в общих стилях

Commit:
fix: moving AddButton and SpendingsInputs to a single component AddForm
feat: add new validation function that shows message under empty input and outlines input, add addCost function

baklanovairina commented on src/components/AddForm/index.jsx:
```diff
@@ -0,0 +1,31 @@
+import "./style.scss";
+import AddButton from "../UI/AddButton";
+import SpendingsInput from "../UI/SpendingsInput";
```
назови просто Input лио CustomInput
это же переиспользуемый компонент

baklanovairina commented on src/components/AddForm/index.jsx:
```diff
@@ -0,0 +1,31 @@
+import AddButton from "../UI/AddButton";
+import SpendingsInput from "../UI/SpendingsInput";
+
+const AddForm = ({ handleChangeInput, cost, addCost, errorInputs }) => {
```
```
const AddForm = ({ 
  handleChangeInput, 
  cost, 
  addCost, 
  errorInputs 
}) => {
```

baklanovairina commented on src/components/AddForm/index.jsx:
```diff
@@ -0,0 +1,31 @@
+
+const AddForm = ({ handleChangeInput, cost, addCost, errorInputs }) => {
+  return (
+    <div className="add-field">
```
className="add-form"

baklanovairina commented on src/components/AddForm/index.jsx:
```diff
@@ -0,0 +1,31 @@
+import "./style.scss";
+import AddButton from "../UI/AddButton";
```
просто Button либо CustomButton

baklanovairina commented on src/components/AddForm/index.jsx:
```diff
@@ -0,0 +1,31 @@
+        textType="number"
+        errorInputs={errorInputs}
+      />
+      <AddButton addCost={addCost} buttonText="Добавить" />
```
type?

handleClickButton={addCost} либо actionButton={addCost}

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,53 @@
+      errorInputs.pop();
+    });
+    checkIfEmpty(['whereSpent', 'howMuchSpent']);
+    if (errorInputs.length === 0) {
```
отступы делай, не пиши сплошной код

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,53 @@
+    checkIfEmpty(['whereSpent', 'howMuchSpent']);
+    if (errorInputs.length === 0) {
+      localStorage.setItem(`spending-${cost.id}`, JSON.stringify(cost));
+      setCost({ ...cost, whereSpent: '', howMuchSpent: '', id: uid() });
```
```
setCost({ 
  ...cost, 
  whereSpent: '', 
  howMuchSpent: '', 
  id: uid() 
});
```

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,53 @@
+
+  const checkIfEmpty = (names) => {
+    names.forEach(name => {
+      if (cost[name]  === '') {
```
не корректная валидация, у тебя одно поле вообще чисто числовым должно быть и проверять не введено ли отрицательное число либо 0

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,53 @@
+        setErrorInputs([ ...errorInputs ]);
+      }
+    })
+  }
```
(;) игноришь

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,53 @@
+  }
+
+  return (
+    <div className="container">
```
почему container? Посмотри как называется твой компонент

baklanovairina commented on src/components/MainPage/style.scss:
```diff
@@ -0,0 +1,22 @@
+  }
+}
+
+body {
```
это общие стили, выноси их в главный файл со стилями

baklanovairina commented on src/components/UI/AddButton/style.scss:
```diff
@@ -0,0 +1,19 @@
+  margin-top: 25px;
+
+  &:hover {
+    cursor: pointer;
```
общие стили

Commit:
fix: rename AddButton to CustomButton, rename SpendingsInput to CustomInput, change validation function, put global styles in global folder, rename App.css to App.scss

Commit:
fix: rename AddButton to CustomButton, rename SpendingsInput to CustomInput, change validation function, put global styles in global folder, rename App.css to App.scss

Commit:
idk

Commit:
feat: add TotalPrice component

Commit:
feat: add costs-container

Commit:
feat: add delete function

Commit:
feat: add edit function

Commit:
Merge branch 'edit-function' into validation

Commit:
fix: change components&class namings, little change for howMuchSpent input validation, fix code style
feat: add helpers folder with convertDate function

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+  const [totalSum, setTotalSum] = useState(0);
+
+  const handleChangeInput = (name, value) => {
+    setCost({ ...cost, [name]: value, whenSpent: Date.now() });
```
так то это поле whenSpent: Date.now()
должно создавать при добавлении расхода, а не при изменении какого либо поля

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+
+  const confirmEdit = (cost) => {
+    if (costToEdit.whenSpent
+      && costToEdit.whenSpent
```
costToEdit.whenSpent дважды??

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+      }
+      
+      sortBySum(allCosts);
+      setCostToEdit({});
```
почему {}?

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+  }
+
+  const cancelEdit = (costCancel) => {
+    for (let i = 0; i < allCosts.length; i++) {
```
вообще вместо этих циклов лучше использовать метод find
это даже будет выглядеть красивее и меньше кода

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+  }
+
+  const sortBySum = (array) => {
+    setAllCosts(array.sort((a, b) => a.howMuchSpent - b.howMuchSpent));
```
мутации!

baklanovairina commented on src/components/MainPage/index.jsx:
```diff
@@ -0,0 +1,170 @@
+        totalSum={totalSum}
+      />
+      <TotalPrice totalSum={totalSum} />
+      <div className="costs-container">
```
откуда costs-container, когда компонент называется main page

baklanovairina commented on src/components/MainPage/style.scss:
```diff
@@ -0,0 +1,30 @@
+  }
+}
+
+body {
```
все еще общий стили, выноси их в главный файл со стилями

baklanovairina commented on src/components/UI/CustomButton/index.jsx:
```diff
@@ -0,0 +1,15 @@
+const CustomButton = ({ actionButton, buttonText }) => {
+  return (
+    <button
+      className="add-button"
```
не add-button а custom-button
нейминг класса должен соответствовать неймингу компонента, в котором этот класс используется

baklanovairina commented on src/components/EditCost/index.jsx:
```diff
@@ -0,0 +1,61 @@
+  cancelEdit
+}) => {
+  return (
+    <div className="cost">
```
что здесь делают классы другого компонента?

baklanovairina commented on src/App.scss:
```diff
@@ -0,0 +1,59 @@
+  height: 45px;
+}
+
+.cost {
```
именнованный класс ледит в общих стилях? 
Он явно принадлежит компоненту Cost

Commit:
fix: rename compnents class names, moving some styles to the global styles file, fix tabs, replace cycles with findIndex function
feat: add adaptive part of website

Commit:
fix: change logic on checking costIndex return value
